Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -82,21 +82,26 @@ public CategoryController(ICategoryRepository categoryRepository, IStringConstan
         /// Delete API to remove a Category
         ///</summary>
         /// <param name="categoryId">The id of the Category to delete.</param>
-        /// <returns>No content(204) response if id found else not found(404) response</returns>
+        /// <returns>Bad request(400) response If questions are associated to the Category else no content(204) response if id found else not found(404) response</returns>
         [HttpDelete("{categoryId}")]
         public async Task<IActionResult> DeleteCategory([FromRoute] int categoryId)
         {
             if (!ModelState.IsValid)
             {
                 return BadRequest(ModelState);
             }
-            Category category = await _categoryRepository.GetCategoryByIdAsync(categoryId);
-            if (category != null)
+            if (!await _categoryRepository.IsCheckCategoryInQuestionsById(categoryId))
```

Comment:
There is no point of checking category id in question model until you first check it in category table itself.